### PR TITLE
Added a mock solr for itests

### PR DESCRIPTION
**Pull request by emmberk**

### CODE REVIEW

**Comment by vinamartin** [comment]
✏️ Probably not necessary for your PR but we should make sure our app still runs even if we can't immediately connect to solr. 
We establish the connection in `ApplicationConfiguration` and if we can't I think our app fails to start.

**Comment by emmberk** [comment]
rebased

File: multi-int-store/src/test/java/com/connexta/multiintstore/MultiIntStoreIntegrationTest.java [hunk 1]
```diff
@@ -20,17 +20,54 @@
...
+import org.springframework.test.context.junit4.SpringRunner;
 import org.springframework.test.web.servlet.MockMvc;
 import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
 import org.springframework.test.web.servlet.setup.MockMvcBuilders;
 import org.springframework.web.context.WebApplicationContext;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.wait.strategy.Wait;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
```
Comment (vinamartin) [inline multi-int-store/src/test/java/com/connexta/multiintstore/MultiIntStoreIntegrationTest.java]: ❓ Are you using the random port? If you were, you'd be injecting it somewhere in this class which I don't see.

**Comment by emmberk** [comment]
rebased

**Comment by emmberk** [comment]
build now

**Comment by emmberk** [comment]
The CI build failed because of a known issue downloading CVEs.

File: multi-int-store/src/test/java/com/connexta/multiintstore/MultiIntStoreIntegrationTest.java [hunk 2]
```diff
@@ -20,17 +20,54 @@
...
+import org.testcontainers.containers.wait.strategy.Wait;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+@ContextConfiguration(initializers = MultiIntStoreIntegrationTest.Initializer.class)
+@EnableConfigurationProperties
+public class MultiIntStoreIntegrationTest {
+
+  @ClassRule
+  public static final GenericContainer solr =
```
Comment (Kjames5269) [inline multi-int-store/src/test/java/com/connexta/multiintstore/MultiIntStoreIntegrationTest.java]: ✏️ what was the reasoning to move this outside the abstract class?

**Comment by emmberk** [comment]
rebased

File: multi-int-store/src/main/java/com/connexta/multiintstore/common/DataRetriever.java [hunk 3]
```diff
@@ -6,48 +6,32 @@
...
 
-  private static final Logger LOGGER = LoggerFactory.getLogger(DataRetriever.class);
+  @NotEmpty private final String callbackAcceptVersion;
 
-  private final ApplicationConfiguration config;
-
-  @Autowired
-  public DataRetriever(ApplicationConfiguration config) {
-    this.config = config;
+  public DataRetriever(@NotEmpty final CallbackAcceptVersion callbackAcceptVersion) {
```
[the remark this answers is not in the document]
Comment (emmberk) [reply multi-int-store/src/main/java/com/connexta/multiintstore/common/DataRetriever.java]: Good catch. `@NotEmpty` can be used for Strings, Collections, Maps, and Arrays. I forgot to update the annotation after changing these configs values to be contained in a class. I updated the annotations for the constructor arguments to be `@NotNull`.

File: multi-int-store/src/test/java/com/connexta/multiintstore/MultiIntStoreIntegrationTest.java (hunk 2 above)
Comment (emmberk) [reply multi-int-store/src/test/java/com/connexta/multiintstore/MultiIntStoreIntegrationTest.java]: I wanted to be really clear about the difference between `MultiIntStoreIntegrationTest` and `MultiIntStoreMockSolrIntegrationTest`. I was thinking it'd be confusing why `MultiIntStoreMockSolrIntegrationTest` didn't also extend `MultiIntStoreIntegrationTestContainers`.

**Comment by emmberk** [comment]
build now

**Comment by emmberk** [comment]
rebased

build now

**Comment by jhunzik** [comment]
`./gradlew clean build && ./gradlew dependencyCheckAnalyze` passed and the hero was successful. ✅

**Pull request merged**